src/rgw/rgw_perf_counters.cc

```diff
@@ -192,6 +202,38 @@ void tinc(const CountersContainer &counters, int idx, ceph::timespan amt) {
 
 } // namespace rgw::op_counters
 
+namespace rgw::persistent_topic_counters {
+
+const std::string rgw_topic_counters_key = "rgw_topic";
+
+PerfCounters* new_perf_counter(const std::string& name, CephContext *cct) {
+  PerfCountersBuilder pcb(cct, name, l_rgw_topic_first, l_rgw_topic_last);
+  add_rgw_topic_counters(&pcb);
+  PerfCounters* new_counters = pcb.create_perf_counters();
+  cct->get_perfcounters_collection()->add(new_counters);
+
+  return new_counters;
+}
+
+
+CountersManager::CountersManager(const std::string& topic_name, CephContext *cct) {
+  const std::string topic_key = ceph::perf_counters::key_create(rgw_topic_counters_key, {{"Topic", topic_name}});
+  topic_counters = std::unique_ptr<PerfCounters>(new_perf_counter(topic_key, cct));
+  cct = cct;
+}
+
+void CountersManager::set(int idx, uint64_t v) {
+  if (topic_counters) {
```

Comment:
`topic_counters` won't ever be null here, will it?

Reply:
correct